### DISCUSSION TITLE
Filter for items created today

### DISCUSSION THREAD

**Discussion by Marshmallow3535** [pocketbase/pocketbase#2794] [Q&A] [answered]
Hi, In javascript I would write the filter to compare the current date and the date + 1 day. However im not sure if this is possible with the scripting language. Is it possible to create a server side filter that only returns records that were created today based on there created attrivute. 

Thanks in advanced!

**Comment by ganigeorgiev** [comment, marked as answer]
>  Is it possible to create a server side filter that only returns records that were created today based on there created attrivute.

There is no special field for targeting "today" in the API rules, only `@now`.
I've added to the roadmap to consider for v0.18/v0.19 eventually adding support for `@todayStart`, `@todayEnd`, `@weekStart`, `@weekEnd`, `@monthStart`, `@monthEnd`, `@yearStart`, `@yearEnd`.

Depending on your use case, one possible workaround could be to create a [View collection](https://pocketbase.io/docs/collections/#view-collection) with a custom SQLite query, for example:

```sql
select
   id,
   created,
   -- ... list all other collection fields
from example
where created >= date() and created < date('now', '+1 day')
```

_(for all SQLite date functions you can check https://www.sqlite.org/lang_datefunc.html)_

**Reply by Marshmallow3535** [reply]
Thanks that’s helpful, I’ll just add client side filtering or use pocketbase as a framework. Why did you decide that custom queries only work for views?

**Reply by ganigeorgiev** [reply]
> Why did you decide that custom queries only work for views?

Because it would be too complicated and my primary use case doesn't require it. There are just too many edge cases if we allow the user to specify the base collection list and view queries and it would be either very limited or will require extensive validations for select fields, joins, builtin functions, aggregations, etc. and at least for now I don't think I'll have the time to work on something like that.
